op-batcher/batcher/driver.go

```diff
@@ -542,92 +542,138 @@ func (l *BatchSubmitter) receiptsLoop(wg *sync.WaitGroup, receiptsCh chan txmgr.
-			// We'd probably hit this error right after startup, so a short shutdown duration should suffice.
+			l.Log.Error("SetMaxDASize RPC method unavailable on endpoint, shutting down. Either enable it or disable throttling.",
+				"endpoint", endpoint, "err", err)
+
+			// We have a strict requirement that all endpoints must have the SetMaxDASize endpoint, and shut down if this RPC method is not available
 			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 			defer cancel()
 			// Call StopBatchSubmitting in another goroutine to avoid deadlock.
 			go func() {
-				// Always returns nil. An error is only returned to expose this function as an RPC.
 				_ = l.StopBatchSubmitting(ctx)
 			}()
 			return
 		} else if err != nil {
-			l.Log.Error("SetMaxDASize rpc failed, retrying.", "err", err)
+			l.Log.Error("SetMaxDASize RPC failed for endpoint, retrying.", "endpoint", endpoint, "err", err)
 			retryTimer.Reset(retryInterval)
 			return
 		}
+
 		if !success {
-			l.Log.Error("Result of SetMaxDASize was false, retrying.")
+			l.Log.Error("Result of SetMaxDASize was false for endpoint, retrying.", "endpoint", endpoint)
 			retryTimer.Reset(retryInterval)
+			return
 		}
-	}
 
-	cachedPendingBytes := int64(0)
+		l.Log.Debug("Successfully set max DA size on endpoint",
+			"endpoint", endpoint,
+			"max_tx_size", maxTxSize,
+			"max_block_size", maxBlockSize)
+	}
 
 	for {
 		select {
-		case pendingBytes, ok := <-pendingBytesUpdated:
+		case _, ok := <-throttleSignal:
 			if !ok {
 				// If the channel was closed, this is our signal to exit
-				l.Log.Info("throttlingLoop returning")
+				l.Log.Info("Endpoint throttling loop shutting down", "endpoint", endpoint)
 				return
 			}
-			updateParams(pendingBytes)
-			cachedPendingBytes = pendingBytes
-		case <-l.ActiveSeqChanged:
-			updateParams(cachedPendingBytes)
+			updateParams()
 		case <-retryTimer.C:
-			updateParams(cachedPendingBytes)
+			updateParams()
 		}
 	}
 }
 
+// throttlingLoop acts as a distributor that spawns individual throttling loops for each endpoint
+// and fans out the pending bytes updates to each endpoint
+func (l *BatchSubmitter) throttlingLoop(wg *sync.WaitGroup, pendingBytesUpdated chan int64) {
+	defer wg.Done()
+	l.Log.Info("Starting DA throttling loop")
+	updateChans := make([]chan struct{}, len(l.Config.ThrottlingEndpoints))
+
+	innerWg := sync.WaitGroup{}
+
+	for i, endpoint := range l.Config.ThrottlingEndpoints {
+		updateChans[i] = make(chan struct{}, 1)
+		innerWg.Add(1)
+		go l.singleEndpointThrottler(&innerWg, updateChans[i], endpoint)
+		l.Log.Info("Started throttling loop for endpoint", "endpoint", endpoint)
+	}
+
+	for pb := range pendingBytesUpdated {
+		l.mutex.Lock()
+		l.throttling = uint64(pb) > l.Config.ThrottleThreshold
+		if l.throttling {
+			l.Log.Warn("Throttling loop: pending bytes above threshold, sending throttling updates", "pending_bytes", pb, "threshold", l.Config.ThrottleThreshold, "throttle", l.throttling)
```

Comment:
I realize that we actually _always_ send a throttle update, so that we also continuously un-throttle. So maybe we should rephrase this log to say that the "sequencers will be throttled" or so?